docs/api/media.rst

```diff
@@ -185,6 +187,8 @@ Supported Handler Types
 .. autoclass:: falcon.media.URLEncodedFormHandler
     :no-members:
 
+.. _custom-media-handler-type:
+
```

Comment:
This was the correct place. The docs for the `custom-media-handler-type` could probably use an example similar to https://falcon.readthedocs.io/en/stable/api/media.html#replacing-the-default-handlers or a link to them.
An example from the previous comment:
```py
extra_handlers = {
    'application/my-media-format: MyMediaFormatHandler()
}

app = falcon.App()

app.req_options.media_handlers.update(extra_handlers)
app.resp_options.media_handlers.update(extra_handlers)
```

Probably a link is better since we avoid repetition and maybe we can also mention it in the examples there that custom media handlers can be added that way.

It's not really the focus of this pr, but I was reading it while reviewing it and I think it could be improved by this change. If an user is linked to it directly without having read the previous part, it does not know how to use the custom handler.

Reply:
Ah, that's a fair point.
I'll link to [Replacing the Default Handlers](https://falcon.readthedocs.io/en/latest/api/media.html#replacing-the-default-handlers) both from this tutorial (i.e., explaining what to actually do with that `json_handler`) and from the custom handler type.

Reply:
Done.